### PR TITLE
feat(actions): add quickfix action

### PR DESCRIPTION
**Pull request by williamboman**
This is a very crucial workflow for me personally, figured I'd open a PR
instead of an issue since I've written the code already - hope you don't mind.

Wdyt?

(will remain in draft until I've tested this enough)

### CODE REVIEW

**Comment by DNLHC** [comment]
Cool! Adding quickfix support was in my plans sadly haven't had much time lately to work on new features
I've briefly tested your PR, works pretty well so far

File: lua/glance/init.lua [hunk 1]
```diff
@@ -242,6 +242,33 @@ Glance.actions = {
...
+          end_lnum = end_lnum,
+          col = item.col,
+          end_col = end_col,
+          text = item.full_text,
+        })
+      end
+    end
+    vim.fn.setqflist(qf_items, 'r')
+    Glance.actions.close()
+    vim.cmd.copen()
```
Comment (williamboman) [inline lua/glance/init.lua:334]: I noticed the README states `>= 0.7` support, vim.cmd indexing was added in `0.8`. Will change this to be 0.7-compatible.

**Comment by williamboman** [comment]
Hey @DNLHC. So I've ran this for a while and it seems to work fine for the most part. I have however run into situations where the qflist is populated with items that miss a filename/bufnr - I've been unable so far to look into why this is happening. I don't think it's quite ready for merge (lmk if you prefer me to close PR and re-open later when it's actually ready)

**Comment by DNLHC** [comment]
> (lmk if you prefer me to close PR and re-open later when it's actually ready)

Hey, i don't mind keeping the PR open.
There is an internal breaking change in the main branch introduced in the multibyte support commit.
These changes will accommodate the new `items` api.

```diff
diff --git a/lua/glance/init.lua b/lua/glance/init.lua
index d21cc8b..ce37a40 100644
--- a/lua/glance/init.lua
+++ b/lua/glance/init.lua
@@ -246,21 +246,13 @@ Glance.actions = {
     local qf_items = {}
     for _, group in pairs(glance.list.groups) do
       for _, item in ipairs(group.items) do
-        local end_lnum = vim.tbl_get(item, 'finish', 'line')
-        if end_lnum then
-          end_lnum = end_lnum + 1
-        end
-        local end_col = vim.tbl_get(item, 'finish', 'character')
-        if end_col then
-          end_col = end_col + 1
-        end
         table.insert(qf_items, {
           bufnr = item.bufnr,
           filename = item.filename,
-          lnum = item.lnum,
-          end_lnum = end_lnum,
-          col = item.col,
-          end_col = end_col,
+          lnum = item.start_line + 1,
+          end_lnum = item.end_line + 1,
+          col = item.start_col + 1,
+          end_col = item.end_col + 1,
           text = item.full_text,
         })
       end
```

> I have however run into situations where the qflist is populated with items that miss a filename/bufnr

Hmm strange that shouldn't happen, I will take some time tomorrow to test it

**Comment by williamboman** [comment]
I'm still experiencing issues with it not producing line:col pairs for entries. I'll revisit this now

edit: Ah nvm, it was due to 7551af1683ebb4ecd76dc01c881ca2b7ce597fce.

**Comment by carlosflorencio** [comment]
I'm waiting for this to be merged as well, thanks for the work @williamboman!

**Comment by DNLHC** [comment]
Thank you

**Pull request merged**